galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/TestRunLifecycleStatus.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.framework;
+
+import java.util.*;
+
+import org.apache.commons.lang3.EnumUtils;
+
+public enum TestRunLifecycleStatus {
+    FINISHED("finished"),
+    BUILDING("building"),
+    GENERATING("generating"),
+    RUNNING("running"),
+    RUNDONE("rundone"),
+    UP("up"),
+    STARTED("started"),
+    PROVSTART("provstart"),
+    ENDING("ending")
+    ;
+
+    
+    private String value ;  
+
+    private TestRunLifecycleStatus(String value) {
+        this.value = value ;
+    }
+   
+
+    public TestRunLifecycleStatus getStatusFromString(String statusInput) {
+        return valueOf(statusInput);
+    }
+
+    public static boolean isStatusValid(String statusInput){
+        return EnumUtils.isValidEnum(TestRunLifecycleStatus.class, statusInput);
+    }
+
+    @Override
+    public String toString(){
+        return value;
+    }
+
+    /**
+     * @return A string list from a TestRunLifecycleStatus list
+     */
+    public static List<String> convertTestRunLifecycleStausListToStringList (List<TestRunLifecycleStatus> testRunLifecycleStatuses){
+        List<String> returnList = new ArrayList<String>();
+        for (TestRunLifecycleStatus status : testRunLifecycleStatuses){
+            returnList.add(status.value);
+        }
+
+        return returnList;
+    }
+
+      
+    /** 
+     * @return A list of possible status names, as strings
+     */
+    public static List<String> getAll() {
```

Comment:
I would have thought getAll() would return a collection of TestRunLifecycleStatus'es rather than strings.

If we need the list of strings, perhaps getAllValues() or something else would be better ?

Why do we need a list of strings anyway ?

Reply:
So getValues is an in built enum method that returns a list of the enums. I believe we used this to pass a list of strings into an error for output, and made it a method to add extra functionality for future uses. Do you think it would be better to maybe rename it instead?

Reply:
changed to allToStringList(), so it's called as TestRunLifecycleStatus.allToStringList().